### PR TITLE
fix(status): fix release status

### PR DESCRIPTION
**Pull request by li-ang**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes https://github.com/caicloud/quality/issues/417

Reference to #

**Special notes for your reviewer**:

/cc @zoumo  @kdada 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/status/assistants/apps/v1/deployment.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 var gvkDeployment = apps.SchemeGroupVersion.WithKind("Deployment")
 
 func DeploymentAssistant(store store.IntegrationStore, obj runtime.Object) (status.Sentence, error) {
+
```
Comment (kdada) [inline pkg/status/assistants/apps/v1/deployment.go]: Remove this line.

File: pkg/status/assistants/apps/v1/deployment.go [hunk 2]
```diff
@@ -29,11 +30,16 @@ func DeploymentAssistant(store store.IntegrationStore, obj runtime.Object) (stat
...
 	available := dp.Status.AvailableReplicas
 	unavailable := dp.Status.UnavailableReplicas
+	conditions := dp.Status.Conditions
+	observedGeneration := dp.Status.ObservedGeneration
 	switch {
 	case unavailable == 0 && desired == current && desired == updated && desired == available:
 		return status.Available, nil
 	case unavailable > 0 && desired == updated && desired != available:
 		// TODO(kdada): Check wrong pods for more precise verdict
+		if observedGeneration == 1 && len(conditions) > 0 && conditions[len(conditions)-1].Type == apps.DeploymentProgressing {
```
Comment (kdada) [inline pkg/status/assistants/apps/v1/deployment.go]: What's the usage of `observedGeneration`?

**Comment by li-ang** [comment]
caicloud-bot 依旧没 merge，手动 merge

**Pull request merged**